client/src/components/MapAllLocationMarker.jsx

```diff
@@ -6,69 +6,67 @@ export default function AllLocationMarker({
   locations,
   roadtrip,
   onUpdateRoadtripLocations,
+  showAllLocations,
 }) {
-  const addLocation = (event) => {
-    const index = locations.findIndex(
-      (oneLocation) => oneLocation._id === event.target.id
-    );
+  const addLocation = (index) => {
     const newLocation = locations[index];
```

Comment:
You could continue working with the location instead of the index.
```suggestion
  const addLocation = (newLocation) => {
    // const newLocation = locations[index];
```